### PR TITLE
Fix freeze on add title text

### PR DESCRIPTION
**Pull request by mike-spa**
Resolves: #24795 

The problem is two-fold. First: the program was getting stuck on a loop because, after adding a new VBox, it was trying to move the staff type changes by a given tick difference, but the VBox has zero tick so the loop was never progressing. Second: the VBox shouldn't be actually added because one is already there. The reason it was being added is that there is an empty part that _doesn't_ have the title VBox, and adding it to the part was also creating it in the score.

### CODE REVIEW

File: src/engraving/dom/edit.cpp [hunk 1]
```diff
@@ -703,7 +703,9 @@ TextBase* Score::addText(TextStyleType type, EngravingItem* destinationElement)
             for (Score* score : scoreList()) {
                 MeasureBase* titleFrame = score->first();
                 if (!titleFrame || !titleFrame->isVBox()) {
-                    titleFrame = score->insertBox(ElementType::VBOX, titleFrame);
+                    InsertMeasureOptions opt = InsertMeasureOptions();
```
Comment (RomanPudashkin) [inline src/engraving/dom/edit.cpp:706]:
```suggestion
                    InsertMeasureOptions opt;
```

**Review by RomanPudashkin — APPROVED** [review]

File: src/engraving/dom/undo.cpp [hunk 2]
```diff
@@ -2235,6 +2235,9 @@ void InsertRemoveMeasures::insertMeasures()
         for (Staff* staff : score->staves()) {
             Fraction tickStart = fm->tick();
             Fraction tickEnd = lm->endTick();
+            if (tickEnd == tickStart) {
```
Comment (cbjeukendrup) [inline src/engraving/dom/undo.cpp:2238]: Looks like this check (and the creation of `tickStart` and `tickEnd`) could just as well happen outside the loop

**Comment by DmitryArefiev** [comment]
@mike-spa Tested on Win/Mac. I still can **crash** with these steps (and that score):

1. Add Title
2. Add Subtitle
3. Undo All
4. Add Title

https://github.com/user-attachments/assets/34694388-158b-48e8-bea4-3e6c1b67c5b0

**Comment by mike-spa** [comment]
@DmitryArefiev should be fixed!

**Comment by DmitryArefiev** [comment]
@mike-spa Found another one.. Can you check it also?

1. Open that score
2. Open all parts
3. Select 1st measure and activate Note input
4. Add Title (with some letters)
5. Undo all
6. Add Title (with some letters)
7. Click on empty space

https://github.com/user-attachments/assets/246c3bf7-1f96-4a7f-bc6b-cc932a5fe01e

**Comment by mike-spa** [comment]
@DmitryArefiev I'm copying your steps but I really can't reproduce this last one. I've rebased the branch, could you check again with the new build?

**Comment by DmitryArefiev** [comment]
@mike-spa I still can't reproduce with that score.  But also, I found a **score that crashes** with the **similar steps** and the crash **doesn't occur in master**. Please try it:

1. Open [bends.zip](https://github.com/user-attachments/files/17097930/bends.zip)
2. Open all parts
3. In score, Add Title (with some letters)
4. Undo all
5. Add Title (with some letters)
6. Undo all


https://github.com/user-attachments/assets/eb090fd8-5a21-4c59-a9ab-db37f3b9cd6a




[logs.zip](https://github.com/user-attachments/files/17097922/logs.zip)

**Comment by mike-spa** [comment]
Here's the situation. There is this piece of code [from here](https://github.com/musescore/MuseScore/pull/19405/commits/58bf266e8bd8ebb4b27535683cd6acb91f55e0ac#diff-4dae0e75e0f35495b4862f2b2236a55da0caa3218daabe36a848349a635ea2f4R683-R696) which is clearly wrong (for example, it causes an unnecessary frame to be added to the main score in the file from the original issue). Moreover [this line](https://github.com/musescore/MuseScore/pull/19405/commits/58bf266e8bd8ebb4b27535683cd6acb91f55e0ac#diff-4dae0e75e0f35495b4862f2b2236a55da0caa3218daabe36a848349a635ea2f4R692) creates spurious duplicate links between the title frames. In debug builds, this causes an assertion failure which was preventing me to even get to the actual issue I was trying to fix, so I tried to fix this code as well. However, it turns out that creating those wrong duplicate links, by pure coincidence, prevents the crash that @DmitryArefiev found in the last video. That crash has a much deeper and much more complex cause which can't be fixed in a patch release.

So, long story short, I've given up on fixing that piece of code, and left in just the bare minimum solution which fixes the linked issue. This should avoid the crashes. And about that last crash, I'll log a separate issue.

**Comment by DmitryArefiev** [comment]
@mike-spa Got it!

Tested #24795 on Win10, Mac13.6, LinuxUbuntu22.04 - **FIXED**

**Review by DmitryArefiev — APPROVED** [review]
